File: tests/xcalablemp/global-view/loop/C/different_template.c
#include <stdio.h>
#include <xmp.h>
#pragma xmp nodes p(*)
#pragma xmp template t0(0:9)
#pragma xmp template t1(0:9)
#pragma xmp distribute t0(block) onto p
#pragma xmp distribute t1(block) onto p
float a[10];
#pragma xmp align a[j] with t0(j)

int main(void){

#pragma xmp loop on t1(i)
  for(int i=0; i<10; i++)
      a[i] = i;

  if(xmp_node_num() == 1)
    printf("PASS\n");

  return 0;
}

File: tests/XMP/global-view/loop/F/different_template.f90
program main
  include 'xmp_lib.h'
!$xmp nodes p(*)
!$xmp template t0(10)
!$xmp template t1(10)
!$xmp distribute t0(block) onto p
!$xmp distribute t1(block) onto p
  real*4 a(10)
!$xmp align a(i) with t0(i)

!$xmp loop (i) on t1(i)
  do i=1, 10
     a(i) = i
  end do

!$xmp task on p(1)
  write(*,*) "PASS"
!$xmp end task
end program main
